velox/functions/prestosql/benchmarks/ZipBenchMark.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/init/Init.h>
+#include "velox/expression/tests/VectorFuzzer.h"
+#include "velox/functions/Macros.h"
+#include "velox/functions/Registerer.h"
+#include "velox/functions/lib/benchmarks/FunctionBenchmarkBase.h"
+#include "velox/functions/prestosql/VectorFunctions.h"
+
+namespace {
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::functions;
+
+template <bool useVectorized>
+class VectorizedLoop : public exec::VectorFunction {
+ public:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx* context,
+      VectorPtr* result) const override {
+    const vector_size_t numInputArrays = args.size();
+
+    for (auto& arg : args) {
+      VELOX_USER_CHECK_EQ(
+          arg->type()->kind(),
+          TypeKind::ARRAY,
+          "vectorized_loop requires arguments of type ARRAY");
+    }
+
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    std::vector<const vector_size_t*> rawSizes(numInputArrays);
+    std::vector<const vector_size_t*> indices(numInputArrays);
+
+    for (int i = 0; i < numInputArrays; i++) {
+      auto baseVector = decodedArgs.at(i)->base()->as<ArrayVector>();
+      rawSizes[i] = baseVector->rawSizes();
+      indices[i] = decodedArgs.at(i)->indices();
+    }
+
+    vector_size_t resultElementsSize = 0;
```

Comment:
FYI - I am keeping this so as to do an apples to apples comparison between two loops.